lookout/style/format/analyzer.py

```diff
@@ -64,23 +113,50 @@ def analyze(self, ptr_from: ReferencePointer, ptr_to: ReferencePointer,
                 fe = FeatureExtractor(language=lang, **rules.origin_config["feature_extractor"])
                 res = fe.extract_features([file], lines)
                 if res is None:
-                    comment = Comment()
-                    comment.file = file.path
-                    comment.confidence = 100
-                    comment.line = 1
-                    comment.text = "Failed to parse this file"
+                    if self.config["report_parse_failures"]:
+                        comment = Comment()
+                        comment.file = file.path
+                        comment.confidence = 100
+                        comment.line = 1
+                        comment.text = "Failed to parse this file"
+                        comment.append(comment)
+                    self.log.warning("Failed to parse %s", file.path)
                     continue
-                X, y, vnodes_y, _ = res
-                self.log.debug("predicting values for %d samples", len(y))
-                y_pred, winners = rules.predict(X, True)
+                X, y, vnodes_y, vnodes = res
+                X, _ = fe.select_features(X, y)
```

Comment:
it's not necessary to call select_features anymore

Reply:
thanks!

Reply:
ping :)

Reply:
it shows in the original thread. :)
Strange, I think I fixed it... Give me 5 mins.